### PR TITLE
Add extension for turbolinks

### PR DESCRIPTION
**Pull request by Krzysztof-Cieslak**

### CODE REVIEW

File: src/Saturn.Extensions.Turbolinks/Turbolinks.fs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+open System
+open Saturn
+open Microsoft.AspNetCore.Builder
+open Microsoft.AspNetCore.Http
+open System.IO
+open FSharp.Control.Tasks.ContextInsensitive
+open System.Threading.Tasks
+open Giraffe
+
+module TurnolinksHelpers =
```
Comment (baronfel) [inline src/Saturn.Extensions.Turbolinks/Turbolinks.fs]: Tiny typo here: Turnolinks -> Turbolinks

File: src/Saturn.Extensions.Turbolinks/Turbolinks.fs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+type Turbolinks (next: RequestDelegate) =
+
+
+
+
+
+    member __.Invoke(ctx: HttpContext) =
+        task {
+            let ms = new MemoryStream()
```
[the remark this answers is not in the document]
Comment (Krzysztof-Cieslak) [reply src/Saturn.Extensions.Turbolinks/Turbolinks.fs]: Huh, I've basically copy pasted this implementation - https://github.com/TerribleDev/TurboLinks.Net/blob/master/src/TurboLinks.Net/TurboLinks.cs and helpers from https://github.com/imranismail/turbolinks.ex

[the remark this answers is not in the document]
Comment (NinoFloris) [reply src/Saturn.Extensions.Turbolinks/Turbolinks.fs]: /shrug

[the remark this answers is not in the document]
Comment (Krzysztof-Cieslak) [reply src/Saturn.Extensions.Turbolinks/Turbolinks.fs]: After looking at some others implementation I think you are correct, not sure why this C# implementation is different. I'll change that

[the remark this answers is not in the document]
Comment (NinoFloris) [reply src/Saturn.Extensions.Turbolinks/Turbolinks.fs]: I hope you see that also means you only need some functions for detecting when a request is through Turbolinks (isXhr), the others are not useful or valid (js setting text/javascript makes no sense for returning an HTML view, xhr or not)

**Pull request merged**